dimagi/utils/django/email.py

```diff
@@ -30,6 +33,14 @@ def send_HTML_email(subject, recipient, html_content, text_content=None,
                          "\n {}".format(html_content)
             )
 
+    if ga_track:
+        url = "https://www.google-analytics.com/collect?v=1&tid={ga_tid}&cid={ga_cid}&t=event&ec=email".format(
+            ga_tid=settings.ANALYTICS_IDS['GOOGLE_ANALYTICS_ID'],
+            ga_cid=uuid.uuid4().hex)
+        new_content = '<img src="{url}&ea=open"/>\n</body>'.format(url=url)
+        html_content, count = re.subn(r'(.*)</body>', r'\1'+new_content, html_content)
+        if count == 0:
+            raise Exception('Cannot add tracking to HTML Email with no closing body tag')
```

Comment:
Should probably raise `RuntimeError` or some other custom exception. `Exception` should generally not be raised because it is very hard to catch without also catching most other errors as well.

On the other hand, if you think this should never happen then maybe it should be changed to an assert statement? I'm not sure which is correct here, but here are my best practices for using assert vs raise exception: Assertions are used to verify logical consistency of code, and should never be able to be triggered by an end user. If the error can be triggered by an end user then raise an exception.